server.js

```diff
@@ -115,16 +120,283 @@ async function initServer(member, alias) {
+
+        var tokenRequest = Token.createTransferTokenRequest(requestData.amount, requestData.currency)
+            .setToAlias(alias)
+            .setToMemberId(member.memberId())
+            .setDescription(requestData.description)
+            .setRedirectUrl(redirectUrl)
+            .setCSRFToken(csrfToken)
+            .setRefId(refId)
+            .setExecutionDate(executionDate.toISOString().split("T")[0])
+            .addTransferDestination(destination);
+
+        var request = await member.storeTokenRequest(tokenRequest)
+        var requestId = request.id;
+        var tokenRequestUrl = Token.generateTokenRequestUrl(requestId);
+
+        res.redirect(302, tokenRequestUrl);
+    });
+
+    app.post('/future-dated-popup', urlencodedParser, async function (req, res) {
+        var redirectUrl = req.protocol + '://' + req.get('host') + '/redeem-future-dated-popup';
+        var destination = {
+            sepa: {
+                iban: 'bic',
+                bic: 'DE16700222000072880129'
+            },
+            customerData: {
+                legalNames: ['merchant-sample-js']
+            }
+        };
+
+        var requestData = req.body;
+        var refId = Token.Util.generateNonce();
+        var csrfToken = Token.Util.generateNonce();
+        req.session.csrfToken = csrfToken;
+
+        //Setting the execution date of payment for after 2 days
+        var executionDate = new Date();
+        executionDate.setDate(new Date().getDate() + 1);
+        var tokenRequest = Token.createTransferTokenRequest(requestData.amount, requestData.currency)
+            .setToAlias(alias)
+            .setToMemberId(member.memberId())
+            .setDescription(requestData.description)
+            .setRedirectUrl(redirectUrl)
+            .setCSRFToken(csrfToken)
+            .setRefId(refId)
+            .setExecutionDate(executionDate.toISOString().split("T")[0])
+            .addTransferDestination(destination);
+
+        var request = await member.storeTokenRequest(tokenRequest)
+        var requestId = request.id;
+        var tokenRequestUrl = Token.generateTokenRequestUrl(requestId);
+
+        res.status(200).send(tokenRequestUrl);
+    });
+
+    // Endpoint for transferring, called by client side after user approval
+    app.get('/one-step-payment', async function (req, res) {
+        var transferDestination = {
+                sepa: {
+                    iban: 'DE16700222000072880129',
+                    bic: '123456'
                 },
+            customerData: {
+                legalNames: ['merchant-sample-js']
+            }
+        };
+
+        var destination = {
+            account: {
+                sepa: {
+                    iban: 'DE16700222000072880129',
+                    bic: '123456'
+                }
             },
+            customerData: {
+                legalNames: ['merchant-sample-js']
+            }
         };
+        var bankId = "stet-cmci";
```

Comment:
@priyanshu-ap the bank id is not supposed to be hardcoded in the code. Why are we doing that.

Reply:
@JugrajSripalJain we raised the same concern with you to not use any bank-specific values to our samples. But you mentioned it's not an issue because we are using only test bank credentials. You can refer our previous comments on these Jira tickets https://tokenio.atlassian.net/browse/AP-96 and also can you please let us know how  to proceed with this now for one-step support